modules/plugin/imagepyramid/src/main/java/org/geotools/gce/imagepyramid/ImageLevelsMapper.java

```diff
@@ -0,0 +1,222 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2016, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.gce.imagepyramid;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Properties;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.geotools.data.DataUtilities;
+import org.geotools.factory.Hints;
+import org.geotools.gce.imagemosaic.ImageMosaicReader;
+
+/**
+ * Parse imagePyramid property files and setup the mapping to provide the proper 
+ * {@link ImageMosaicReader} for the required image choice. 
+ * Starting with 16.x we are supporting ImagePyramids of ImageMosaic with internal 
```

Comment:
I thought this was supposed to be backported to 15.x too? No need to cite the version in the javadoc anyways